### PR TITLE
Change how prices are calculated fixes #167

### LINKED ISSUE #167: At 0 Charisma, Items Sell for 0 Gold
**Bug Summary**
If you have your charisma set to 0, you won't get any gold for selling items to the shop.

**Expected behavior**
I would expect to get at least some gold.

**Observed Behaviour**
I get 0 gold.

**Steps To Reproduce**
1.  Start game
2. Make your Charisma 0
3. Make your way to the shop
4. Sell an item to the shop

**Environment**
- Version: Current Master
- OS: Linux, OSX, Windows
- Browser: Chrome, Safari, Firefox, Edge

---

### PR DESCRIPTION
**Pull request by Matteas-Eden**
GitHub Issue (If applicable): #167 

## PR Type

What kind of change does this PR introduce?

- Bugfix
- Enhancement

## What is the current behavior?

A player who has 0 Charisma sells items for 0 gold.

## What is the new behavior?

Item prices are now calculated according to a range of percentages of the base value of an item. There are three percentages involved with the calculation: `MIN_SELL_PERCENT`, `MID_SELL_PERCENT` and `MAX_SELL_PERCENT`.

The sell price of an item will be a percentage of the base price in the range `MIN_SELL_PERCENT` to `MID_SELL_PERCENT`.

The buy price of an item will be a percentage of the base price in the range `MID_SELL_PERCENT` to `MAX_SELL_PERCENT`.

The exact percentage value within those ranges is determined by the player's charisma modifier.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current upstream repo
- [ ] Added new tests to the test suite for the feature
- [ ] Tested the full test suite, and ensured that the feature does not break current build.
- [x] Docs have been added/updated which fit (for bug fixes / features)
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal)
- [ ] Checked licensing of Frameworks/Libraries (if applicable) 

## Other information

Internal Issue (If applicable): Closes #167.

### CODE REVIEW

File: src/utils/calculate-prices.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    MIN_PRICE_PERCENT,
+    MID_PRICE_PERCENT,
+    MAX_PRICE_PERCENT,
+} from '../config/constants';
+
+// Calculates buy and sell price of an item
+// Params
+//  min: minimum percentage of base price that item will sell for
+//  mid: max sell percentage, min buy percentage
+//  max: maximum percentage of base price that item can be bought for
```
Comment (KimberleyEvans-Parker) [inline src/utils/calculate-prices.js]: I think these comments are outdated

**Review by KimberleyEvans-Parker — CHANGES_REQUESTED** [review]
Works great :D just fix the comments please

**Review by Pyxxil — CHANGES_REQUESTED** [review]
Note that, as #180 was closed before this, we will also need to update how we retrieve the sell price for the item being viewed (src/components/view-item/index.js line 195)

**Review by KimberleyEvans-Parker — DISMISSED** [review]
seems to work well for me :)

**Review by Pyxxil — APPROVED** [review]
Looks fine

**Pull request merged**